File: 数据结构/树/num257.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<string> paths;
    vector<string> binaryTreePaths(TreeNode* root) {
        string path = "";
        dfs(root, path);
        return paths;
    }
    void dfs(TreeNode* root, string path){
        if(!root) return;
        // 叶子节点
        else if(!root->left && !root->right){
            path += to_string(root->val);
            paths.push_back(path);
            return;
        }
        // 递归
        else{
            path += to_string(root->val) + "->";
            dfs(root->left, path);
            dfs(root->right, path);
        }
    }
};

File: 数据结构/树/num257.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def binaryTreePaths(self, root):
        """
        :type root: TreeNode
        :rtype: List[str]
        """
        paths = []
        def dfs(root, path):
            if not root:
                return
            elif not(root.left or root.right):
                path += str(root.val)
                paths.append(path)
                return
            else:
                path += str(root.val) + "->"
                dfs(root.left, path)
                dfs(root.right, path)

        path = ""
        dfs(root, path)
        return paths